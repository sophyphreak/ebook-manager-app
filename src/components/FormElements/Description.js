import React, { Component } from 'react';
import TextAreaInput from './MainElements/TextAreaInput';

const Description = (props) => (
  <TextAreaInput
    {...props}
  />
);

export default Description;