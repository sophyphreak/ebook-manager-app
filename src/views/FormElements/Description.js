import React from 'react';
import TextAreaInput from './MainElements/TextAreaInput';

const Description = ({ label, description, onDescriptionChange, ...props }) => (
  <TextAreaInput
    label={label}
    value={description}
    onChange={onDescriptionChange}
    {...props}
  />
);

export default Description;
