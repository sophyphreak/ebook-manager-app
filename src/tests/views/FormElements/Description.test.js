import React from 'react';
import { shallow } from 'enzyme';
import Description from '../../../views/FormElements/Description';

let wrapper;

beforeEach(() => {
  wrapper = shallow(<Description />);
});

test('should render Description correctly', () => {
  expect(wrapper).toMatchSnapshot();
});
