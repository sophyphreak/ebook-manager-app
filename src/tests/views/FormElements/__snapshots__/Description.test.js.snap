// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Description correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Description />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Description:*",
        <textarea
          onChange={undefined}
          type="text"
          value={undefined}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      "Description:*",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "onChange": undefined,
          "type": "text",
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "textarea",
      },
    ],
    "type": "label",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Description:*",
          <textarea
            onChange={undefined}
            type="text"
            value={undefined}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        "Description:*",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onChange": undefined,
            "type": "text",
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "textarea",
        },
      ],
      "type": "label",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
